### DISCUSSION TITLE
Audio stops playing in background - Is there a workaround for iOS 13+?

### DISCUSSION THREAD

**Discussion by JeremyCaltranz** [ionic-team/capacitor#3446] [Q&A]
Hello, I have an existing App which is a music App similar to Pandora. I have recently moved the App to Capacitor and cannot publish it due to one unsolvable iOS issue.

When the App goes into background mode the audio stops playing after :15 seconds. The audio element is in a HTML audio tag.

This feature is essential to our App and I do not think Capacitor or Cordova has a solution. I am hoping someone in the community can suggest something that might work.

Here is the error that Xcode throws: [ProcessSuspension] Background task expired while holding WebKit ProcessAssertion

So I need some sort of workaround.

These two plugins do not work: cordova-plugin-background-mode or cordova-plugin-background-mode-fixed

I also tried to use the Capacitor Background Task API. However, this too did not work and threw the same ProcessAssertion error above. I followed the example in the documentation where there is a date function in a for loop that continuously runs in the background. 

Here is a screen shot of my Background Modes in Xcode. I've tried all the combinations of checking/unchecking and nothing here seems to make a difference.

![Screen Shot 2020-08-20 at 12 58 28 PM](https://user-images.githubusercontent.com/1087810/90819865-5fbf9280-e2e5-11ea-80bf-a4f681c3c032.png)

Any thoughts or suggestions are greatly appreciated.

Thanks, Jeremy

**Comment by jcesarmobile** [comment]
This is an iOS 13 bug, it was fixed on iOS 14 beta 2, but according to a comment, it was reintroduced.
Should probably be fixed once iOS 14 is final, but will never work on iOS 13 devices, so you should probably move to a plugin that uses native reproduction instead of audio tags.

The webkit bug report for reference https://bugs.webkit.org/show_bug.cgi?id=203293

**Reply by Topiya** [reply]
Can you please suggest any plugin or a way to play audio using URL as you mentioned here '_native reproduction_' ?

**Comment by msoni11** [comment]
This fixes the issue https://github.com/photonstorm/phaser/pull/5706/commits/1a0b0beaaea99ac7e38545ceca46f096711b0f19

**Comment by thomasklemm** [comment]
@JeremyCaltranz Is this issue still happening for you? Evaluating Ionic and Capacitor for an application that needs to be able to keep playing audio when the screen gets locked.

I also came across the [Ionic Native Media Capacitor/Cordova Plugin](https://ionicframework.com/docs/native/media), which looks like it could help with playing audio files (recording is not needed in our use case).

**Comment by JeremyCaltranz** [comment]
This code implementation (below) stopped working with Capacitor 3. It worked fine in version 2 with the plugin.  

When I searched for percentage of iOS devices running > version 14 it was 13%. 

So for those running a lower OS under 14 I check and force an upgrade. 

Here is the code that I developed for anyone interested:

x = new Media(playing_url,
        function () {
            console.log("Audio Completed Callback");
        },
        function (err) {
            console.log("Audio Error: " + err);
        },
        mediaStatus
    );

    x.play(); 

Then I created a mediaStatus function: 

function mediaStatus(status) {
    if(status === 2) {
        console.log('Media is Playing');
    }

    else if(status === 3) {
        console.log('Media is Paused');
    }

    else if(status === 4) {
        console.log('Media Stopped');
    }
}




Jeremy Callahan,
Callahan & Associates Inc.
www.jeremycallahan.com <http://www.jeremycallahan.com/>
415-250-6168
Twitter @jeremycallahan
Schedule a call with me at: https://calendly.com/jeremy_callahan/15min/ <https://calendly.com/jeremy_callahan/15min/>
